app/src/utils/get-nested-field/get-nested-field.ts

```diff
@@ -0,0 +1,36 @@
+import { useFieldsStore, useRelationsStore } from '@/stores';
+import i18n from '@/lang';
+import { Relation } from '@/types';
+const fieldsStore = useFieldsStore();
+const relationsStore = useRelationsStore();
+
+export default function getNestedField(item: any, path: string, collection: string): any {
+	const splitPath = path.split('.');
+	const pathBefore = [];
+	while (item) {
+		const key = splitPath.shift();
+		if (!key) {
+			break;
+		}
+		pathBefore.push(key);
+		item = item[key];
+		if (Array.isArray(item)) {
+			const field = fieldsStore.getField(collection, pathBefore.join('.'));
+			if (field.type === 'translations') {
+				const relations: Relation[] = relationsStore.getRelationsForField(field.collection, field.field);
+				const translationsRelation = relations.find(
+					(relation) => relation.one_collection === field.collection && relation.one_field === field.field
+				);
+				if (!translationsRelation) {
+					break;
+				}
+				const languagesCodeField = translationsRelation.junction_field;
+				if (!languagesCodeField) {
+					break;
+				}
+				item = item.find((el) => el[languagesCodeField] === i18n.locale) || item[0];
+			}
+		}
```

Comment:
Instead of having a hardcoded resolver inside the system getter, I think I'd prefer a setup where a display can signal up what filters to use for the relational item in presentation, and then render it based on that. That allows us to create a `translations` display that shows it in the current language, but (more importantly) also gives us the ability to have the `related-values` display render a single item in-line based on the same filter setup (so instead of showing "15 Items" it can show "_Hello, World!_ and 14 others" or something along those lines